File: set1/fixed_xor.py
import codecs

hex1 = bytes.fromhex('1c0111001f010100061a024b53535009181c')
hex2 = bytes.fromhex('686974207468652062756c6c277320657965')

def fixed_xor(h1: bytes, h2: bytes) -> str:
  if len(h1) != len(h2):
    print('Oops the lengths are not the same')
    return h1
  # bh1 = bytes.fromhex(h1)
  # bh2 = bytes.fromhex(h2)
  res = [d1 ^ d2 for d1, d2 in zip(h1, h2)]

  return codecs.encode(bytes(res), 'hex').decode()

if __name__ == "__main__":
  print(fixed_xor(hex1, hex2))

File: rust/set1/src/fixed_xor.rs
pub fn fixed_xor(h1: &str, h2: &str) -> String {
    if h1.len() != h2.len() {
        return "Error".to_string();
    }

    let h1_bytes = match hex::decode(h1) {
        Ok(bytes) => bytes,
        Err(_) => return "Error".to_string(),
    };

    let h2_bytes = match hex::decode(h2) {
        Ok(bytes) => bytes,
        Err(_) => return "Error".to_string(),
    };

    let mut res = Vec::new();
    let mut i = 0;
    while i < h1_bytes.len() {
        res.push(&h1_bytes[i] ^ &h2_bytes[i]);
        i += 1;
    }

    return hex::encode(res);
}
